actions/callback.go

```diff
@@ -65,7 +65,8 @@ type KeyFunc func(info ProtectorInfo, retry bool) (*crypto.Key, error)
 // passphrase sources or just relays the callback for raw sources.
 func getWrappingKey(info ProtectorInfo, keyFn KeyFunc, retry bool) (*crypto.Key, error) {
 	// For raw key sources, we can just use the key directly.
-	if info.Source() == metadata.SourceType_raw_key {
+	if info.Source() == metadata.SourceType_raw_key || 
+	   info.Source() == metadata.SourceType_raw_keyEncodedStr {
```

Comment:
This shouldn't really be a separate type of protector, but rather just another way of specifying the raw key for a raw key protector.

Reply:
Exactly It's one thing , a raw key from different sources.. so we could say the sources of the raw key source. 

Last changes makes it easy to add more sources for raw key. For may case,, I need to do some sort of remote encryption/unlocking mechanism... maybe over ssh or any key management server api.. still don't know best way to do so. Pam might be the best place for the unlocking part but not the encryption itself...